[Bug]: double type is not supported in oceanbase parser module

### Check Before Asking

- [X] Please check the issue list and confirm this bug is encountered for the first time.
- [X] Please try full text in English and attach precise description.

### Environment

```markdown
v0.1.0
```


### Fast Reproduce Steps

sql = """
        SELECT Winner FROM table_11621915_1 WHERE Purse > 964017.2297960471 AND Date_ds = "may 28"
        """
sql = Utils.remove_sql_text_affects_parser(sql)
result = oceanbase_parser.parse(sql)

### Actual Behavior

E           SyntaxError: The current version does not support this SQL 56 (.) 
E            select winner from table_11621915_1 where purse > 964017.2297960471 and date_ds = "may 28"

### Expected Behavior

_No response_

### Other Information

_No response_

--- a/parser/oceanbase_parser/lexer.py
+++ b/parser/oceanbase_parser/lexer.py
@@ -16,7 +16,7 @@ from parser.oceanbase_parser.reserved import *
 
 reserved = sorted(set(presto_tokens).difference(presto_nonreserved))
 
-tokens = ['INTEGER', 'DECIMAL', 'NUMBER',
+tokens = ['INTEGER', 'NUMBER', 'DOUBLE',
           'IDENTIFIER', 'DIGIT_IDENTIFIER',
           'QUOTED_IDENTIFIER', 'BACKQUOTED_IDENTIFIER',
           'STRING', 'PERIOD',
@@ -54,6 +54,13 @@ t_QM = r'\?'
 t_CONCAT = r'\|\|'
 
 t_ignore = ' \t'
+
+
+def t_DOUBLE(t):
+    r"""(\d+(?:\.\d*)?(?:[eE][+-]?\d+)?|\d*(?:\.\d+)(?:[eE][+-]?\d+)?)"""
+    if 'e' in t.value or 'E' in t.value or '.' in t.value:
+        t.type = 'DOUBLE'
+    return t
 
 
 def t_INTEGER(t):

--- a/parser/oceanbase_parser/lexer.py
+++ b/parser/oceanbase_parser/lexer.py
@@ -60,6 +60,8 @@ def t_DOUBLE(t):
     r"""(\d+(?:\.\d*)?(?:[eE][+-]?\d+)?|\d*(?:\.\d+)(?:[eE][+-]?\d+)?)"""
     if 'e' in t.value or 'E' in t.value or '.' in t.value:
         t.type = 'DOUBLE'
+    else:
+        t.type = "INTEGER"
     return t
 
 

--- a/parser/oceanbase_parser/parser.py
+++ b/parser/oceanbase_parser/parser.py
@@ -1059,9 +1059,9 @@ def p_quoted_identifier(p):
 
 
 def p_number(p):
-    r"""number : DECIMAL
+    r"""number : DOUBLE
                | INTEGER"""
-    if p.slice[1].type == "DECIMAL":
+    if p.slice[1].type == "DOUBLE":
         p[0] = DoubleLiteral(p.lineno(1), p.lexpos(1), p[1])
     else:
         p[0] = LongLiteral(p.lineno(1), p.lexpos(1), p[1])

--- a/test/parser/test_parser_dml.py
+++ b/test/parser/test_parser_dml.py
@@ -441,6 +441,14 @@ INSERT IGNORE INTO bumonitor_risk_process_context (gmt_create, gmt_modified, row
         result = oceanbase_parser.parse(sql)
         assert isinstance(result, Statement)
 
+    def test_float_type(self):
+        sql = """
+        SELECT Winner FROM table_11621915_1 WHERE Purse > 964017.2297960471 AND Date_ds = "may 28"
+        """
+        sql = Utils.remove_sql_text_affects_parser(sql)
+        result = oceanbase_parser.parse(sql)
+        assert isinstance(result, Statement)
+
 
 if __name__ == '__main__':
     unittest.main()

--- a/test/parser/test_parser_dml.py
+++ b/test/parser/test_parser_dml.py
@@ -441,7 +441,7 @@ INSERT IGNORE INTO bumonitor_risk_process_context (gmt_create, gmt_modified, row
         result = oceanbase_parser.parse(sql)
         assert isinstance(result, Statement)
 
-    def test_float_type(self):
+    def test_double_type(self):
         sql = """
         SELECT Winner FROM table_11621915_1 WHERE Purse > 964017.2297960471 AND Date_ds = "may 28"
         """

